### ISSUE TITLE
Missing transaction leads to SequelizeConnectionAcquireTimeoutError.

### ISSUE THREAD

**Issue by royalpinto** [sequelize/sequelize#11995]
###### What are you doing?
When some Sequelize calls within a transaction weren't using the parent transaction. So something like:
```
sequelize.transaction(async transaction => {
  await Model.create({ name: 'Test' }, { transaction })
  await DifferentModel.findAll() // missing transaction
})
```

It seems that these calls block the transaction for a larger amount of time and other transactions fail with SequelizeConnectionAcquireTimeoutError due to lack of free connections.

Ref: https://github.com/sequelize/sequelize/issues/10858#issuecomment-549817032

I don't see any reason why it should be kept open for a longer time when a promise is already fulfilled and the transaction can be committed and connection can be released immediately.

**Comment by royalpinto** [comment]
After analyzing, I found that it's not really an issue.

#### What's happening:
The transaction block ( `sequelize.transaction(async transaction => {}` ) waits for all the promises inside it to finish before committing and releasing the connection. While the line which is missing transaction waits for a new transaction to be available. Hence deadlock.

It's easy to imagine with only a single connection. The transactioned connection waits for the non-transactioned line to finish while the non-transactioned line waits for the connection to be available which is taken by the transaction. Hence eventually it will give up and throw `SequelizeConnectionAcquireTimeoutError`.

It would have been nice if there is a way to find missing transaction lines like this.

**Comment by aecorredor** [comment]
I think I'm running into the same issue as you. But, I'm still not quite getting why this would happen. Let's say that I don't really care about the transaction in the second sequelize call, maybe because it's not important in the logic that should make the transaction roll back. Wouldn't it make sense for me not to have to pass the transaction to that call? @royalpinto 

The docs don't really talk much about this (at least for v5). They do have an example where they show a transaction within a transaction and they pass `null` to the `transaction` property of one of the calls: https://sequelize.org/v5/manual/transactions.html in the concurrent/parallel transactions section.

**Comment by anniestudent** [comment]
I run into this a few days ago and this result seemingly can be affected by the config setting `pool.max`. If `pool.max` is 5, firing 5 requests that involve partial transaction operations simultaneously will see `SequelizeConnectionAcquireTimeoutError` at server side and `socket hang up` at client side after waiting for a while. Firing 4 requests will be ok tho.

**Comment by saalihou** [comment]
Would this be solved by passing `null` as a transaction to the 2nd call that does not depend on the transaction ? (e.g. https://sequelize.org/master/manual/transactions.html#:~:text=create(%7B%20name%3A%20%27Alice%27%20%7D)%3B%0A%7D)%3B-,Concurrent/Partial%20transactions,-You%20can%20have)

**Comment by hexpunk** [comment]
I know I'm necro-posting, but I wanted to share my story around this in case it helps anyone else.

I had added a transaction to our app's critical path and I started receiving `SequelizeConnectionAcquireTimeoutError`s.  This all seems obvious in retrospect, but I spent the better part of this morning figuring this out.

If I removed the transaction wrapping that part of the code _or_ increased the maximum size of the connection pool (I used `Infinity` for testing purposes 😂) the problem went away.  So I dropped the connection pool down to exactly 1.  It appeared as if the first request was completing just fine, but then the next request would hang until a `SequelizeConnectionAcquireTimeoutError` occurred.  It's as if the existence of the transaction caused the connection to never be returned to the pool.

After a lot of searching Sequelize issues, this issue seemed to be the closest to my problem.  I stopped and read the docs regarding _implicit_ transactions _very_ closely.  I double-checked the app.  We're using a CLS namespace during unit tests, but not in production.  We were previously only using transactions to wrap unit tests and revert the changes at the end.  We weren't using transactions in the actual app code until I recently introduced a few.  But this was the first one in the critical app path and the first one to use implicit transactions.  That's what made it click for me.  I added a namespace to the main app connection and now everything works as expected.
